### ISSUE TITLE
Add read_sdmx convenience method

### ISSUE THREAD

**Issue by javihern98** [bis-med-it/pysdmx#156]
Add the read_sdmx method:

- Supports all readers implemented in io package
- Uses the input processor to use a file path or string buffer or URL
- Infers the format to use from an enumeration that holds the supported formats

**Comment by antonio-olleros** [comment]
As an example of the method

```
def get_dataset(data_path:str|Path, metadata_path:str|Path=None, data_format=Pd) -> "DataSet":
    #Get files format
        1. is_xml, is_csv, is_json
        2. Version
    #Validate files
        1. Can I parse de file with what I got in 1 and 2?
    
    if metadata_path:
        string_metadata, _ = process_string_to_read(metadata_path)
        metadata = read_xml(string_metadata, validate=False)

    

    string_data, _ = process_string_to_read(data_path)
    data = read_xml(string_data, validate=False)

    data.structure = metadata

    return data
```

**Comment by javihern98** [comment]
@sosna Need some guidance on the current status of SDMX-JSON and FusionJSON reading:

- Are data readers implemented?
- Could you provide a small script to read these files and parse the structures?

**Comment by sosna** [comment]
> Are SDMX-JSON data readers implemented?

No, we have JSON readers to support the FMR use case, i.e. reading structural and reference metadata. Coming with generic readers and writers going beyond that use case is scheduled for later this year (it's on my workplan already).

**Comment by javihern98** [comment]
> > Are SDMX-JSON data readers implemented?
> 
> No, we have JSON readers to support the FMR use case, i.e. reading structural and reference metadata. Coming with generic readers and writers going beyond that use case is scheduled for later this year (it's on my workplan already).

I will leave them as NotImplemented in the read_sdmx method then, testing it accordingly
